Assingment_Tving/Assingment_Tving/Network/Endpoint/HomeAPI.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  HomeEndpoint.swift
+//  Assingment_Tving
+//
+//  Created by 이성민 on 2023/05/11.
+//
+
+import Foundation
+
+import Alamofire
+
+enum HomeAPI<T: Codable>: RequestProtocol, ResponseProtocol {
+    
+    case fetchMoviePopular
+    case fetchTVPopular
+    
+    // MARK: - RequestProtocol
+    
+    var path: String {
+        switch self {
+        case .fetchMoviePopular: return "movie/popular"
+        case .fetchTVPopular: return "tv/popular"
+        }
+    }
+    
+    var method: HTTPMethod {
+        switch self {
+        case .fetchMoviePopular: return .get
+        case .fetchTVPopular: return .get
+        }
+    }
+    
+    var headers: HTTPHeaders? {
+        switch self {
+        case .fetchMoviePopular:
+            return ["accept": "application/json",
+                    "Authorization": "Bearer " + Config.apiKey]
+        case .fetchTVPopular:
+            return ["accept": "application/json",
+                    "Authorization": "Bearer " + Config.apiKey]
+        }
+    }
+    
+    // MARK: - ResponseProtocol
+    
+    var type: T.Type {
+        switch self {
+        case .fetchMoviePopular: return T.self
+        case .fetchTVPopular: return T.self
+        }
+    }
```

Comment:
어떤 case가 들어와도 return하는 값이 똑같아서 switch를 안해도 되지 않을까요...?
```suggestion
    var type: T.Type {
        return T.self
    }
```